cli/main.go

```diff
@@ -173,6 +174,12 @@ func (c *DiceDBClient) handleWatchCommand(cmd string, args []string) {
 	go c.watchCommand(baseCmd, toArgInterface(args[1:])...)
 }
 
+func (c *DiceDBClient) handleWatchModeExit() {
+	c.subCancel()
+	c.subscribed = false
+	c.subType = ""
```

Comment:
Would suggest this to be moved to `case <-c.subCtx.Done():`
The `defer` function in the `watchCommand` is not immediately getting executed when `cancel()` is called and meanwhile the `subCtx.Done()` channel will unblock and exit the loop. We can reset `subscribed` and `subType` Immediately on Cancel as part of exit itself.

Reply:
But, cancellation message is processed by this `watchCommand` goroutine asynchronously, right? So `LivePrefix` method can be executed even before we reset `subscribed` to false.
This state on which the `LivePrefix` depends should be reset synchronously.